### ISSUE TITLE
No checkpoints saved when training keras_conformer

### ISSUE THREAD

**Issue by iamweiweishi** [TensorSpeech/TensorFlowASR#180]
I am trying to train a Keras-Conformer model, but after several epochs, I found that no checkpoints saved in predefined filepath. 
If I used 'train_ga_subword_conformer.py' instead of 'train_keras_subword_conformer.py', the checkpoints could be saved successfully. 

params:

    running_config:
        batch_size: 2
        accumulation_steps: 2
        num_epochs: 5
        outdir: trained_models/
        log_interval_steps: 10
        eval_interval_steps: 100
        save_interval_steps: 100
        checkpoint:
          filepath: trained_models/checkpoints/{epoch:02d}.h5
          save_best_only: True
          save_weights_only: False
          save_freq: epoch
        states_dir: trained_models/states
        tensorboard:
          log_dir: trained_models/tensorboard
          histogram_freq: 1
          write_graph: True
          write_images: True
          update_freq: epoch
          profile_batch: 2

But I could only find tensorboard and states in trained_models directory.

Is this a tensorflow bug?

Tensorflow: pip install tensorflow-gpu==2.3.0;
Python: 3.6

**Comment by usimarit** [comment]
@iamweiweishi Did you create the checkpoints folder before training? The keras checkpoint callback doesn't auto create the folder so you have to do it yourself. You can load the states from states_dir then store to .h5 format.

**Comment by iamweiweishi** [comment]
@usimarit The checkpoints folder does exist. I can find ckpts are saved into 'states' successfully. But when the training is finished, the 'states' folder is cleared, and no checkpoints are saved in 'filepath'.

**Comment by usimarit** [comment]
@iamweiweishi can you try setting `save_best_only: False` and `save_weights_only: True`? Maybe setting `save_best_only` will look for best the `monitor` (if this is not defined then fallback to `monitor='val_loss'`) but for tensorflow_asr version 0.x, monitor has to be defined as `monitor: val_rnnt_loss` for this to work), so it cannot find the best `val_rnnt_loss`, that's why it doesn't save checkpoints.
FYI: I changed the name of model's loss metric to the default 'loss' and 'val_loss' in the new version 1.x so we don't have to define `monitor: val_rnnt_loss` or 'monitor: val_ctc_loss` anymore 😄

**Comment by iamweiweishi** [comment]
@usimarit 
Thank you. Maybe that is the problem. I will upgrade TensorFlowASR in a few day. 💯
